File: CodeTest/Python/SWEA/D1_2056.py
MONTH_DAY = {
    '01':31,
    '02':28,
    '03':31,
    '04':30,
    '05':31,
    '06':30,
    '07':31,
    '08':31,
    '09':30,
    '10':31,
    '11':30,
    '12':31
}

testCase = int(input())

for i in range(1, testCase+1):
    num = input().rstrip()
    
    year = num[:4]
    month = num[4:6]
    day = num[6:]

    if int(month)<1 or int(month)>12:
        year = '-1'

    if year != '-1' and MONTH_DAY[month] < int(day):
        year = '-1'

    if year == '-1':
        print(f'#{i} -1')
    else:
        print(f'#{i} {year}/{month}/{day}')

File: CodeTest/Java/SWEA/D1_2056.java
import java.util.*;

public class D1_2056 {
    final static int[] MONTH_DAY = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int testCase = sc.nextInt();

        String year, month, day;

        for(int i=1; i<=testCase; i++) {
            String str = sc.next();
            year = str.substring(0, 4);
            month = str.substring(4, 6);
            day = str.substring(6, 8);

            if(Integer.parseInt(month)>12 || month.equals("00")) {
                year = "-1";
            }

            if(!year.equals("-1") && (MONTH_DAY[Integer.parseInt(month)] < Integer.parseInt(day) || day.equals("00"))) {
                year = "-1";
            }

            if(year.equals("-1")) {
                System.out.println("#" + i + " " + year);
            } else {
                System.out.println("#" + i + " " + year + "/" + month + "/" + day);
            }
        }
    } 
}
